src/program.cpp

```diff
@@ -347,7 +347,7 @@ void program::finalize()
 template <class T>
 std::string classify(T x)
 {
-    switch(std::fpclassify(x))
+    switch(std::fpclassify(static_cast<double>(x)))
```

Comment:
This breaks for a subnormal float conversion to double. It won't stay subnormal anymore. Thanks.